applications/ShallowWaterApplication/custom_modelers/mesh_moving_modeler.cpp

```diff
@@ -114,12 +117,41 @@ void MeshMovingModeler::SetupModelPart()
                 new_cond_nodes.push_back(moving_model_part.pGetNode(r_node.Id()));
             }
             auto new_cond = r_cond.Clone(++last_cond_id, new_cond_nodes);
+            old_to_new_cond_id[r_cond.Id()] = last_cond_id;
             new_cond->SetProperties(moving_model_part.pGetProperties(r_cond.GetProperties().Id()));
             aux_array_with_condition_pointers.push_back(new_cond);
         }
     }
     moving_model_part.AddConditions(aux_array_with_condition_pointers.begin(), aux_array_with_condition_pointers.end());
 
+    // SubModelParts
+    std::vector<std::size_t> nodes_ids;
+    std::vector<std::size_t> elements_ids;
+    std::vector<std::size_t> conditions_ids;
+    for (const auto& sub_model_part : fixed_model_part.SubModelParts())
+    {
+        auto& dest_sub_model_part = moving_model_part.CreateSubModelPart(sub_model_part.Name());
+
+        for (const auto& node : sub_model_part.Nodes()) {
+            if (node.Is(TO_COPY)) {
+                nodes_ids.push_back(node.Id());
+            }
+        }
+        for (const auto& element : sub_model_part.Elements()) {
+            if (element.Is(TO_COPY)) {
+                elements_ids.push_back(old_to_new_elem_id[element.Id()]);
+            }
+        }
+        for (const auto& condition : sub_model_part.Conditions()) {
+            if (condition.Is(TO_COPY)) {
+                conditions_ids.push_back(old_to_new_cond_id[condition.Id()]);
+            }
```

Comment:
FYI this could be done in parallel with corresponding reductions

AFAIK the [AccumReduction](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/reduction_utilities.h#L189) could be used for this

Reply:
thanks, interesting

Reply:
AFAIK it is not possible to write a lambda with return inside if but without else (there will be an implicit return 0 which breaks the code at runtime)
BTW, it is not critical since it is done only once and the following statements are serial (`dest_sub_model_part.AddNodes(nodes_ids)`)